### ISSUE TITLE
Bug: `textarea` inputs in Safari stop accepting input if certain lines are deleted

### ISSUE THREAD

**Issue by iansan5653** [facebook/react#28360]
This is a strange one that I honestly couldn't believe wasn't caused by our app, but it turns out I can reproduce it in a minimal fresh React app as well: if you have a `<textarea>` input and you select and delete any line other than the first, second, or last, you cannot type in that input anymore until you leave and refocus it. This is only reproducible in Safari.

React version: 18.2.0
Safari version: 17.3
MacOS version: Sonoma 14.3

## Steps To Reproduce

1. Render a `<textarea>` (controlled or uncontrolled) with four or more lines of initial content
2. Select a line other than the first, second, or last
3. Delete that line by pressing <kbd>Delete</kbd>
4. Attempt to keep typing

Link to code example: https://github.com/iansan5653/textarea-safari-bug

Screen recording of the bug (based on running the linked repo):

https://github.com/facebook/react/assets/2294248/12cc3ece-5f0d-4856-8420-a080fa2d569f

## The current behavior

Upon deleting the line, the line cannot be typed into anymore. Moving the caret to another line allows typing, but moving it back to the deleted line does not. Sometimes an errant character appears in the deleted line. Blurring (moving focus away from) the input and refocusing it makes the whole input interactive again, but does not restore lost input.

## The expected behavior

The input should remain full interactive as normal.

**Comment by blghns** [comment]
I am seeing this issue too https://jsfiddle.net/bilgehansolo/0vmL5cse/ here is the same code on a fiddle. Also tried passing the value as a defaultValue instead of children, it still had the same problem.

**Comment by keithamus** [comment]
Instead of setting the value inside the node, I refactored it to use a `useEffect` and set the actual DOM nodes `.value`. The bug seems to not be present in this case. https://jsfiddle.net/nkrLvm7w/

**Comment by blghns** [comment]
I wonder if this is a React bug or a ReactDOM bug.

**Comment by Nedomas** [comment]
I don't think this is related to deleting. I get the same problem if I start entering text on a previously empty line, but one that has content in lines after it.

For example, enter this into the textarea:
```
1
2

4
```

Then click on the missing line (where 3 would be) and try to start typing - it won't work.

**Comment by iansan5653** [comment]
Oh interesting. Maybe a better title than would be "`textarea` inputs in Safari stop accepting input on certain empty lines" 🤔

**Comment by eimerlegui** [comment]
> Oh interesting. Maybe a better title than would be "`textarea` inputs in Safari stop accepting input on certain empty lines" 🤔

Test

**Comment by iansan5653** [comment]
bump 🙃

**Comment by Nedomas** [comment]
Bump, I think this is still relevant.

**Comment by iansan5653** [comment]
bump bump

**Comment by Nedomas** [comment]
Still relevant.

**Comment by Nedomas** [comment]
This is still relevant.

**Comment by Nedomas** [comment]
Still relevant.

**Comment by Unreliable22** [comment]
I tackled that challenge and identified a practical solution https://jboss-arquillian.github.io/pEub39GUXAHXz8aVJS0KLnM4Uex542sc8EV3bP7LTN4MM7M9N/
